Marlin/src/module/temperature.cpp

```diff
@@ -532,7 +532,7 @@ int Temperature::getHeaterPower(int heater) {
     uint8_t fanDone = 0;
     for (uint8_t f = 0; f < COUNT(fanPin); f++) {
       #ifdef ARDUINO
-        int8_t pin = pgm_read_byte(&fanPin[f]);
+        pin_t pin = pgm_read_byte(&fanPin[f]);
```

Comment:
This whole #ifdef block could be replaced with the following:
```
pint_t pin;
memcpy_P(&pin, &fanPin[f], sizeof(pin_t));
```

Reply:
I'd rather minimize the AVR code size by keeping the #ifdef block as is.  It also is much clearer as to what is happening.

I did verify that the LPC1768 needs to get rid of the pgm_read_byte.

Reply:
Interesting. I didn't think it would really make the code much larger since it's still a variable and a function call. Maybe the `sizeof(pin_t)` would add a few bytes?